ansible/roles/ocp4-workload-integreatly/files/fuse/create-instance.yml

```diff
@@ -0,0 +1,124 @@
+  shell: oc get catalogsource rhmi-registry-cs -n {{ _instance_namespace }} -o jsonpath='{.status.connectionState.lastObservedState}'
+  register: _action_lastObservedState
+  until:  _action_lastObservedState.stdout == "READY"
+  delay: 5
+  retries: 100
+
+- set_fact:
+    _instance_operator_group_file: "{{ ocp4_workload_integreatly_tmp_dir }}/instance-{{ _instance_user }}-opertor-group.yaml"
+
+# Create operator group
+- name: Process operator group template for {{_instance_namespace}}
+  template:
+    src: "{{ocp4_workload_integreatly_fuse_template_operator_group}}"
+    dest: "{{_instance_operator_group_file}}"
+  vars:
+    _instance_namespace: _instance_namespace
+
+- name: Create operator group
+  shell: oc create -f {{_instance_operator_group_file}} -n {{ _instance_namespace }}
+  register: _action_create_operatorgroup
+  failed_when: _action_create_operatorgroup.rc != 0 and 'AlreadyExists' not in _action_create_operatorgroup.stderr
+
+- set_fact:
+    _instance_subscription_file: "{{ ocp4_workload_integreatly_tmp_dir }}/instance-{{ _instance_user }}-subscription.yaml"
+
+# Create subscription
+- name: Process subscription template for {{ _instance_namespace }}
+  template:
+    src: "{{ ocp4_workload_integreatly_fuse_template_subscription }}"
+    dest: "{{ _instance_subscription_file }}"
+  vars:
+    _instance_user: _instance_user
+    _instance_namespace: _instance_namespace
+
+- name: Create subscription
+  shell: oc create -f {{ _instance_subscription_file }} -n {{ _instance_namespace }} 
+  register: _action_create_subscription
+  failed_when: _action_create_subscription.rc != 0 and 'AlreadyExists' not in _action_create_subscription.stderr
+
+# Waiting for installplan to become available
+- name: waiting for installplan to become available in {{_instance_namespace}}
+  shell: oc get installplans -n {{_instance_namespace}} --ignore-not-found=true | grep -v NAME | wc -l
+  register: _action_installplan_exists
+  until:  _action_installplan_exists.stdout == "1"
+  delay: 5
+  retries: 10
+
+# Approve install plan
+- name: Approve installPlan
+  shell: "oc patch installplan $(oc get installplans -n {{_instance_namespace}} | grep -v NAME | awk '{print $1}') -n {{_instance_namespace}} --type='json' -p '[{'op': 'replace', 'path': '/spec/approved', 'value': true}]'"
+  register: _action_patch_installplan
+  failed_when: _action_patch_installplan.rc != 0
+
+# Create the Syndesis CR
+- set_fact:
+    _instance_syndensis_cr_file: "{{ ocp4_workload_integreatly_tmp_dir }}/instance-{{ _instance_user }}-syndensis-cr.yaml"
+
+- name: "Process the template for the Syndesis CR for {{_instance_namespace}}"
+  template:
+    src: "{{ ocp4_workload_integreatly_fuse_template_syndensis_cr }}"
+    dest: "{{ _instance_syndensis_cr_file }}"
+  vars:
+    _instance_namespace: _instance_namespace 
+    _instance_user: _instance_user  
+
+
+- name: Create the Syndesis CR
+  shell: oc create -f {{ _instance_syndensis_cr_file }} -n {{ _instance_namespace }}
```

Comment:
I think we need to use the `k8s` module here instead of `shell`, see [1] for examles

[1] https://github.com/redhat-cop/agnosticd/pull/1688/files#diff-4fa294fd658d75cbf6cb35bac2b4090cR43

Reply:
Seems a bit verbose but lets do it for the fun of it . :smile: